lib/ansible/modules/cloud/amazon/ecs_task.py

```diff
@@ -220,6 +242,14 @@ def format_network_configuration(self, network_config):
                 except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
                     self.module.fail_json_aws(e, msg="Couldn't look up security groups")
             result['securityGroups'] = groups
+        if network_config['assign_public_ip'] is not None:
```

Comment:
Understood -- I used ecs_service as my point of reference for trying to maintain code consistency. I honestly did not do a full review of the entire source directory. This was a fairly rushed hack to fix the bug I reported because it really is a stopping issue for using FARGATE with awsvpc (which is the standard way most people use FARGATE). There are multiple files in cloud/amazon that need this work. Perhaps flag this cleanup as an issue for devel ?

Reply:
I wouldn't want to 'flag this as an issue for devel'.  My suggestion is as follows:

Add a new method ecs_api_handles_assign_public_ip
```
    def ecs_api_handles_assign_public_ip(self):
        from distutils.version import LooseVersion
        return LooseVersion(botocore.__version__) >= LooseVersion('1.8.4')
```

Around line 373 add a quick test:

```
    if module.params['network_config'] and module.params['network_config']['assign_public_ip'] is not None and not service_mgr.ecs_api_handles_assign_public_ip():
        module.fail_json(msg='botocore needs to be version 1.8.4 or higher to use assign_public_ip')
```

This in turn simplifies the logic here to

```
        if network_config['assign_public_ip'] is not None:
            if network_config['assign_public_ip'] is True:
￼                result['assignPublicIp'] = "ENABLED"
￼            else:
￼                result['assignPublicIp'] = "DISABLED"
```


I know we've got a lot of modules which "aren't up to code", this is part of the reason we're trying to be a little more reserved about new code.